File: BasicLevel_C++/1018. 锤子剪刀布 (20).cpp
#include <iostream>
using namespace std;
int main() {
    int n;
    cin >> n;
    int jiawin = 0, yiwin = 0;
    int jia[3] = {0}, yi[3] = {0};
    for (int i = 0; i < n; i++) {
        char s, t;
        cin >> s >> t;
        if (s == 'B' && t == 'C') {
            jiawin++;
            jia[0]++;
        } else if (s == 'B' && t == 'J') {
            yiwin++;
            yi[2]++;
        } else if (s == 'C' && t == 'B') {
            yiwin++;
            yi[0]++;
        } else if (s == 'C' && t == 'J') {
            jiawin++;
            jia[1]++;
        } else if (s == 'J' && t == 'B') {
            jiawin++;
            jia[2]++;
        } else if (s == 'J' && t == 'C') {
            yiwin++;
            yi[1]++;
        }
    }
    cout << jiawin << " " << n - jiawin - yiwin << " " << yiwin << endl << yiwin << " " << n - jiawin - yiwin << " " << jiawin << endl;
    int maxjia = jia[0] >= jia[1] ? 0 : 1;
    maxjia = jia[maxjia] >= jia[2] ? maxjia : 2;
    int maxyi = yi[0] >= yi[1] ? 0 : 1;
    maxyi = yi[maxyi] >= yi[2] ? maxyi : 2;
    char str[4] = {"BCJ"};
    cout << str[maxjia] << " " << str[maxyi];
    return 0;
}

File: BasicLevel_Golang/1018. 锤子剪刀布 (20).go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func findMax(a [3]int) int {
	var max, index int
	for k, v := range a {
		if v > max {
			max = v
			index = k
		}
	}
	return index
}

func main() {
	var (
		n             int
		jiawin, yiwin int
		jia, yi       [3]int
		str           = "BCJ"
	)
	// 用fmt读取输入会有测试点超时
	inputReader := bufio.NewReader(os.Stdin)
	fmt.Scanln(&n)
	for i := 0; i < n; i++ {
		s, _ := inputReader.ReadString('\n')
		t := strings.Fields(s)
		if t[0] == "B" && t[1] == "C" {
			jia[0]++
		} else if t[0] == "B" && t[1] == "J" {
			yi[2]++
		} else if t[0] == "C" && t[1] == "B" {
			yi[0]++
		} else if t[0] == "C" && t[1] == "J" {
			jia[1]++
		} else if t[0] == "J" && t[1] == "B" {
			jia[2]++
		} else if t[0] == "J" && t[1] == "C" {
			yi[1]++
		}
	}
	jiawin = jia[0] + jia[1] + jia[2]
	yiwin = yi[0] + yi[1] + yi[2]
	fmt.Println(jiawin, n-jiawin-yiwin, yiwin)
	fmt.Println(yiwin, n-jiawin-yiwin, jiawin)
	fmt.Printf("%c %c", str[findMax(jia)], str[findMax(yi)])
}
